torch/csrc/jit/tensorexpr/kernel.h

```diff
@@ -129,6 +131,18 @@ class TORCH_API TensorExprKernel {
 
   void bindInput(const torch::jit::Value* input);
 
+  // Captures the information for reduction operation nodes.
+  struct ReductionInfo {
+    std::vector<DimArg> reductionDims;
+    std::vector<DimArg> outputDims;
+    std::vector<size_t> axes;
+    bool keepdim;
+    c10::optional<Dtype> dtype;
+  };
+
+  // Get the reduction info for the given node, based on properties and inputs.
+  ReductionInfo getReductionInfo(const torch::jit::Node* node);
+
```

Comment:
Do these need to be exposed in the header or can they be local to kernel.cpp?

Reply:
Yes, unfortunately. The `getReductionInfo` method needs to be a non-static member of `TensorExprKernel` because it calls `sizesForValue`.